kork-core/src/main/java/com/netflix/spinnaker/kork/spring/PluginLoader.java

```
@@ -0,0 +1,119 @@
+      addPlugins(props);
+    } catch (IOException e) {
+      throw new PluginLoaderException(e);
+    }
+    addJarsToClassPath(source);
+  }
+
+  private PluginProperties parsePluginConfigs(InputStream inputStream)
+      throws PluginLoaderException {
+    ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    try {
+      PluginProperties pluginConfigs = objectMapper.readValue(inputStream, PluginProperties.class);
+      return pluginConfigs;
+    } catch (IOException e) {
+      throw new PluginLoaderException("Not able to parse plugin configurations: " + e.getMessage());
+    }
+  }
+
+  private InputStream openFromFile() throws IOException {
+    return new FileInputStream(this.filename);
+  }
+
+  public void addPlugins(PluginProperties pluginProperties) throws PluginLoaderException {
+    List<Path> jarPaths = getEnabledJarPaths(pluginProperties);
+    this.urls = jarPathsToUrls(jarPaths);
+  }
+
+  private List<Path> getEnabledJarPaths(PluginProperties pluginProperties)
+      throws PluginLoaderException {
+    try {
+      return pluginProperties.pluginConfigurationList.stream()
+          .filter(isEnabled())
+          .map(PluginProperties.PluginConfiguration::getPath)
+          .map(Paths::get)
+          .collect(Collectors.toList());
+    } catch (Exception e) {
+      throw new PluginLoaderException("Was not able to get plugin jar paths: " + e.getMessage());
+    }
+  }
+
+  private Predicate<PluginProperties.PluginConfiguration> isEnabled() {
+    return p -> p.enabled == true;
+  }
+
+  private List<URL> jarPathsToUrls(List<Path> jars) throws PluginLoaderException {
+    try {
+      return jars.stream().map(this::getUrlfromPath).collect(Collectors.toList());
+    } catch (RuntimeException e) {
+      throw new PluginLoaderException(e);
+    }
+  }
+
+  private URL getUrlfromPath(Path path) {
+    try {
+      return path.toUri().toURL();
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
```

Don't throw RuntimeException. Just don't catch it and let the method above deal with it.

---

if we throw a MalformedURLException we have to catch in this method or in the map function call (its a checked exception). I had cast it to a RuntimeException so it can bubble up to be caught later. I'm thinking of updating loadPlugins should either catch all exceptions or just IOException and RuntimeExceptions. thoughts?

---

I removed the other method that called this and have added this to the `getEnabledJarURLs` method, so we get enabled jars as urls in one fell swoop

---

we looked at https://dzone.com/articles/exception-handling-in-java-streams yesterday, which recommended to use RuntimeException as one way of handling this. I'm not sure I want to go the whole "Either" or "Try" route listed in the doc

---

Throwing a `RuntimeException` is bizarre. You can either make the PluginException unchecked or change the logic.

---

ah I think I see, I'll making PluginException a RuntimeException (so its unchecked), but still catch it in the `getEnabledJarURLs` method